tracer/src/Datadog.Trace/Agent/DiscoveryService/DiscoveryService.cs

```diff
@@ -15,41 +15,45 @@ namespace Datadog.Trace.Agent.DiscoveryService
 {
     internal class DiscoveryService : IDiscoveryService
     {
+        private const string InfoPath = "/info";
         private static readonly string[] SupportedDebuggerEndpoints = new[] { "debugger/v1/input" };
         private static readonly string[] SupportedProbeConfigurationEndpoints = new[] { "v0.7/config" };
+        private static readonly string[] SupportedStatsEndpoints = new[] { "v0.6/stats" };
 
         private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<DiscoveryService>();
 
         private readonly IApiRequestFactory _apiRequestFactory;
         private readonly Uri _agentUri;
 
         private DiscoveryService(
-            Uri agentUri,
             IApiRequestFactory apiRequestFactory)
         {
-            _agentUri = agentUri;
             _apiRequestFactory = apiRequestFactory;
+            _agentUri = apiRequestFactory.GetEndpoint(InfoPath);
         }
 
-        public static string[] AllSupportedEndpoints => SupportedDebuggerEndpoints.Concat(SupportedProbeConfigurationEndpoints).ToArray();
+        public static string[] AllSupportedEndpoints => SupportedDebuggerEndpoints.Concat(SupportedProbeConfigurationEndpoints).Concat(SupportedStatsEndpoints).ToArray();
 
         public string ProbeConfigurationEndpoint { get; private set; }
 
         public string DebuggerEndpoint { get; private set; }
 
+        public string StatsEndpoint { get; private set; }
+
         public string AgentVersion { get; private set; }
 
-        public static DiscoveryService Create(IConfigurationSource configurationSource, IApiRequestFactory apiRequestFactory)
+        public bool ClientDropP0s { get; private set; }
+
+        public static DiscoveryService Create(IApiRequestFactory apiRequestFactory)
         {
-            var exporterSettings = new ExporterSettings(configurationSource);
-            return new DiscoveryService(exporterSettings.AgentUri, apiRequestFactory);
+            return new DiscoveryService(apiRequestFactory);
         }
 
         public async Task<bool> DiscoverAsync()
```

Comment:
I assume this call be called several times (by debugger, by us). Shall we keep the response and don't recall the agent everytime?
Also, what happens if we fail to connect to the agent at startup? A good example is AAS, where we start the agent after the Tracer basically and so we may need a retry logic here, no?

Reply:
+1 , also as I mentioned [here](https://github.com/DataDog/dd-trace-dotnet/pull/2988#discussion_r936170197) it'd probably be good to go beyond that and have `DiscoveryService` query the endpoint periodically and notify of changes (changes could happen both because of an upgrade/downgrade but also from features being enabled/disabled: e.g. Remote Config's endpoint will appear/disappear based on whether it is enabled/disabled in `datadog-agent`'s config yaml.

Reply:
FYI, implemented here: https://github.com/DataDog/dd-trace-dotnet/pull/3154